### ISSUE TITLE
serenity-bdd update: HttpsProxyAgent causes TypeError 

### ISSUE THREAD

**Issue by pauko** [serenity-js/serenity-js#1795]
I'm trying to replay the [Serenity/JS Cucumber Template](https://github.com/serenity-js/serenity-js-cucumber-template.git) (at commit https://github.com/serenity-js/serenity-js-cucumber-template/commit/dcd1deae8afefc1c45034a32eefd613ef44a3ece) behind a corporate HTTP(S) Proxy. The version of `serenity-bdd` used therein is **3.6.1**

When doing `npm install` the `postinstall` script runs `serenity-bdd update`, which crashes with the follow trace:

```
ConfigurationError: UpdateCommandActors encountered a problem when preparing actor "Serenity/JS Updater" for stage
    at Stage.actor (/home/dummy/serenity-js-cucumber-template/node_modules/@serenity-js/core/lib/stage/Stage.js:89:23)
    at Stage.theActorCalled (/home/dummy/serenity-js-cucumber-template/node_modules/@serenity-js/core/lib/stage/Stage.js:67:21)
    ... 3 lines matching cause stack trace ...
    at /home/dummy/serenity-js-cucumber-template/node_modules/yargs/build/index.cjs:1:8993
    at j (/home/dummy/serenity-js-cucumber-template/node_modules/yargs/build/index.cjs:1:4956)
    at _.handleValidationAndGetResult (/home/dummy/serenity-js-cucumber-template/node_modules/yargs/build/index.cjs:1:8962)
    at _.applyMiddlewareAndGetResult (/home/dummy/serenity-js-cucumber-template/node_modules/yargs/build/index.cjs:1:9604)
    at _.runCommand (/home/dummy/serenity-js-cucumber-template/node_modules/yargs/build/index.cjs:1:7231)
Caused by: TypeError: HttpsProxyAgent is not a constructor
    at proxiedHttpsAgent (/home/dummy/serenity-js-cucumber-template/node_modules/@serenity-js/serenity-bdd/lib/cli/io/axiosClient.js:122:12)
    at axiosClient (/home/dummy/serenity-js-cucumber-template/node_modules/@serenity-js/serenity-bdd/lib/cli/io/axiosClient.js:54:15)
    at UpdateCommandActors.actors [as createAxios] (/home/dummy/serenity-js-cucumber-template/node_modules/@serenity-js/serenity-bdd/lib/cli/commands/update.js:41:110)
    at UpdateCommandActors.prepare (/home/dummy/serenity-js-cucumber-template/node_modules/@serenity-js/serenity-bdd/lib/cli/stage/UpdateCommandActors.js:16:57)
    at Stage.actor (/home/dummy/serenity-js-cucumber-template/node_modules/@serenity-js/core/lib/stage/Stage.js:84:35)
    at Stage.theActorCalled (/home/dummy/serenity-js-cucumber-template/node_modules/@serenity-js/core/lib/stage/Stage.js:67:21)
    at Serenity.theActorCalled (/home/dummy/serenity-js-cucumber-template/node_modules/@serenity-js/core/lib/Serenity.js:193:27)
    at actorCalled (/home/dummy/serenity-js-cucumber-template/node_modules/@serenity-js/core/lib/instance.js:207:29)
    at Object.handler (/home/dummy/serenity-js-cucumber-template/node_modules/@serenity-js/serenity-bdd/lib/cli/commands/update.js:47:46)
    at /home/dummy/serenity-js-cucumber-template/node_modules/yargs/build/index.cjs:1:8993 {
  cause: TypeError: HttpsProxyAgent is not a constructor
      at proxiedHttpsAgent (/home/dummy/serenity-js-cucumber-template/node_modules/@serenity-js/serenity-bdd/lib/cli/io/axiosClient.js:122:12)
      at axiosClient (/home/dummy/serenity-js-cucumber-template/node_modules/@serenity-js/serenity-bdd/lib/cli/io/axiosClient.js:54:15)
      at UpdateCommandActors.actors [as createAxios] (/home/dummy/serenity-js-cucumber-template/node_modules/@serenity-js/serenity-bdd/lib/cli/commands/update.js:41:110)
      at UpdateCommandActors.prepare (/home/dummy/serenity-js-cucumber-template/node_modules/@serenity-js/serenity-bdd/lib/cli/stage/UpdateCommandActors.js:16:57)
      at Stage.actor (/home/dummy/serenity-js-cucumber-template/node_modules/@serenity-js/core/lib/stage/Stage.js:84:35)
      at Stage.theActorCalled (/home/dummy/serenity-js-cucumber-template/node_modules/@serenity-js/core/lib/stage/Stage.js:67:21)
      at Serenity.theActorCalled (/home/dummy/serenity-js-cucumber-template/node_modules/@serenity-js/core/lib/Serenity.js:193:27)
      at actorCalled (/home/dummy/serenity-js-cucumber-template/node_modules/@serenity-js/core/lib/instance.js:207:29)
      at Object.handler (/home/dummy/serenity-js-cucumber-template/node_modules/@serenity-js/serenity-bdd/lib/cli/commands/update.js:47:46)
      at /home/dummy/serenity-js-cucumber-template/node_modules/yargs/build/index.cjs:1:8993
}
```

If I trace it back I realize that it's `@serenity-js/serenity-bdd/lib/cli/io/axiosClient.js` at line 122 which creates a `HttpsProxyAgent` object and provokes the `TypeError: HttpsProxyAgent is not a constructor`.

As a work-around I can bypass the problematic lines of code by invoking it like this: `serenity-bdd update --repository http://insecure.repo1.maven.org/maven2/`, which overrides `https://repo1.maven.org/maven2/` from `defaults.ts` so that the Axios Client is configured with a HTTP Proxy Agent rather than HTTPS. 

How can this be fixed in a more sustainable and secure way?

**Comment by jan-molak** [comment]
Thanks for reporting, @pauko, I'll take a look.

**Comment by jan-molak** [comment]
Thanks for reporting the issue, @pauko, I can reproduce it. A fix is on the way.

**Comment by jan-molak** [comment]
Fixed in 3.7.1; please let me know if you encounter any other issues!
